### PR TITLE
Batch stream

### PR DESCRIPTION
**Pull request by tomrf1**

### CODE REVIEW

File: app/services/DynamoChannelTests.scala [hunk 1]
```diff
@@ -47,32 +49,56 @@ class DynamoChannelTests(stage: String) extends StrictLogging {
...
+        .get(tableName)
+        .map(items => items.asScala.toList)
+        .getOrElse(Nil)
 
     }.mapError(error =>
       DynamoPutError(error)
     )
 
-  def getAllTests[T <: ChannelTest[T] : Decoder](channel: Channel): ZIO[Blocking, DynamoGetError, List[T]] =
+  /**
```
Comment (tomrf1) [inline app/services/DynamoChannelTests.scala:73]: @adamnfish any thoughts on `putAllBatched`? The use of an empty stream feels a bit odd, but it gives me all the features I wanted

File: app/services/DynamoChannelTests.scala [hunk 2]
```diff
@@ -47,32 +49,56 @@ class DynamoChannelTests(stage: String) extends StrictLogging {
...
+          logger.info(s"next batch: $nextBatch")
+          putAll(nextBatch).map {
+            case Nil => remainingBatches
+            case unprocessed => unprocessed :: remainingBatches
+          }
+        case (Nil, _) =>
+          logger.info(s"finished batches")
+          ZIO.succeed(Nil)
+      }
+      .map(_ => ()) // on success, the result value isn't meaningful
```
Comment (kelvin-chappell) [inline app/services/DynamoChannelTests.scala:98]:
```suggestion
      .unit // on success, the result value isn't meaningful
```

**Review by kelvin-chappell — APPROVED** [review]
It's a nice solution.  Quite easy to see what's going on 👍

File: app/services/DynamoChannelTests.scala (hunk 2 above)
Comment (tomrf1) [reply app/services/DynamoChannelTests.scala:98]: nice!

**Pull request merged**